core/sphinx/path/path_factory.go

```diff
@@ -0,0 +1,46 @@
+package path
+
+import (
+	"errors"
+	mRand "math/rand"
+	"time"
+
+	"github.com/katzenpost/katzenpost/core/crypto/rand"
+	"github.com/katzenpost/katzenpost/core/pki"
+	"github.com/katzenpost/katzenpost/core/sphinx/constants"
+	"github.com/katzenpost/katzenpost/core/sphinx/geo"
+)
+
+// PathFactory is used to compose Sphinx packet paths.
+type PathFactory struct {
+	rng *mRand.Rand
+}
+
+func NewPathFactory() *PathFactory {
+	return &PathFactory{
+		rng: rand.NewMath(),
+	}
+}
+
+// ComposePath is used to compose a Sphinx packet path. Returns
+// path and round trip time or an error.
+func (d *PathFactory) ComposePath(
+	geo *geo.Geometry,
+	doc *pki.Document,
+	srcMix *[32]byte,
+	dstId []byte,
+	dstMix *[32]byte,
+	surbID *[constants.SURBIDLength]byte,
+	baseTime time.Time,
```

Comment:
does this need `time.Time` or could it be using epochs?

Reply:
path select uses time